src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,325 @@
+            {
+                throw new ArgumentException("The command did not contain any bytes to send.");
+            }
+
+            SendCommand(command.GetBytes());
+
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller.
+        /// </summary>
+        /// <param name="command">Command to send.</param>
+        public void SendCommand(Ssd1351Command command)
+        {
+            SendSPI(new byte[] { (byte)command }, true);
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller.
+        /// </summary>
+        /// <param name="command">Command to send.</param>
+        /// <param name="data">Byte array to send as parameters for the command.</param>
+        public void SendCommand(Ssd1351Command command, byte[] data)
+        {
+
+            SendCommand(command);
+
+            if (data != null && data.Length > 0)
+            {
+                SendSPI(data);
+
+                // detect certain commands that may alter the state of the device. This is done as the 
+                // SPI device cannot read registers from the ssd1351 and so changes need to be captured
+                switch (command)
+                {
+                    // capture changes to the colour depth and colour sequence
+                    case Ssd1351Command.SetRemap:
+                        _colorSequence = (ColorSequence)((data[0] >> 2) & 0x01);
+                        _colorDepth = (ColorDepth)((data[0] >> 6) & 0x03);
+                        break;
+                }
+
+            }
+
+        }
+
+        /// <summary>
+        /// Send a command to the the display controller.
+        /// </summary>
+        /// <param name="data">Byte array to send as a combination of command in the first byte
+        /// and following with parameters for the command.</param>
+        public void SendCommand(byte[] data)
+        {
+            Span<byte> buffer = data;
+
+            if(data == null || data.Length == 0)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            if (data.Length == 0)
+            {
+                throw new ArgumentException("Data array is empty", nameof(data));
+            }
+
+            SendCommand((Ssd1351Command)data[0], buffer.Slice(1).ToArray());
+        }
+
+        /// <summary>
+        /// Send data to the display controller.
+        /// </summary>
+        /// <param name="data">The data to send to the display controller.</param>
+        public void SendData(byte[] data)
+        {
+            const int StackThreshold = 512;
+
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            Span<byte> buffer = data.Length < StackThreshold ? stackalloc byte[data.Length + 1] : new byte[data.Length + 1];
+
+            buffer[0] = 0x40; // Control byte.
+            data.CopyTo(buffer.Slice(1));
```

Comment:
using spans everywhere will also save you from having to do this call.

Reply:
@joperezr 

Not sure about this ? I am essentially adding a byte to the beginning of a byte[] How else could I do this ?

Reply:
I see you are right, yeah this is not super trivial. I guess this extra copy is not ideal but it is fine for now.